ctest.h

```diff
@@ -22,6 +22,12 @@
 #define CTEST_IMPL_FORMAT_PRINTF(a, b)
 #endif
 
+#if defined(__GNUC__) && (__GNUC__ >= 4)
```

Comment:
Clang also supports this attribute, so how about this?
```suggestion
#if defined(__clang__) || (defined(__GNUC__) && (__GNUC__ >= 4))
```

Reply:
Alternatively, a more portable solution may be to put the `write` call in an `if` statement with an empty body, maybe with a comment explaining why we don't care if the call fails.

Reply:
Alternatively, a more portable solution may be to put the `write` call in an `if` statement with an empty body, maybe with a comment explaining why we don't care if the call fails.

Reply:
Wrap write call with if is more simple - no worry about compiler specific.